### PR TITLE
emacsPackages: fix build for a few hundred packages

### LINKED ISSUE #335442: Flip the default value of  `ignoreCompilationError` to `false` for Emacs package
In https://github.com/NixOS/nixpkgs/pull/328573, `ignoreCompilationError` is added to elisp build helpers.  To keep the old behavior, its default value is `true`, which means `nix build` succeeds even if there are native compilation errors.

Flipping the default value of `ignoreCompilationError` to `false` means native compilation errors cause `nix build` to fail.  This is desirable  because native compilation errors usually mean packaging issues or issues in the package itself.

In my test, 12.7% of `emacsPackages` fail to build if `ignoreCompilationError` is set to `false`.  So to flip its default value, we need to fix native compilation for those packages first.

The first and easy part is done in https://github.com/NixOS/nixpkgs/pull/335440.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by jian-lin**
## Description of changes

This PR fixes almost all build failures introduced by https://github.com/NixOS/nixpkgs/pull/343388.

The only remaining build failure is `org-xlatex`, which fails for good reason: `Error: error ("Your Emacs was not built with Xwidget support")`.

It is better to review this PR one commit at a time.

Future work:
- resolve TODOs
- work with upstreams to solve compilation errors hidden by `ignoreCompilationError`

This PR is the final part of https://github.com/NixOS/nixpkgs/issues/335442.  Closes #335442.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix [hunk 1]
```diff
@@ -733,6 +733,9 @@ let
           })
           else super.osx-dictionary;
 
+        # https://github.com/skeeto/at-el/issues/9
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix:736]: Upstream fixup?
https://github.com/skeeto/at-el/commit/0489e15cd0bc2fd9da56a4147f9083c6d816ebb9

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix (hunk 1 above)
Comment (jian-lin) [reply pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix:736]: Upstrem responsed after I created this PR.  I do not have time to check and verify the fix yet.  I will do that later if no one does.

BTW, this is the only pending fix needed to be checked and verified for now.

Comment (jian-lin) [reply pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix:737]: resolved

File: pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix [hunk 2]
```diff
@@ -718,6 +733,812 @@ let
...
+        # missing optional dependencies
+        ssh-tunnels = addPackageRequires super.ssh-tunnels [ self.helm ];
+
+        # https://github.com/brianc/jade-mode/issues/73
+        stylus-mode = addPackageRequires super.stylus-mode [ self.sws-mode ];
+
+        # missing optional dependencies
+        suggest = addPackageRequires super.suggest [ self.shut-up ];
+
+        symex = ignoreCompilationError super.symex; # elisp error
```
Comment (MatrixManAtYrService) [inline pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix:1473]: I[ reported a bug to the symex maintainer](https://github.com/drym-org/symex.el/issues/148#issuecomment-2431312021), but it turns out that they couldn't replicate it.  We managed to fix it anyhow, but it makes me wonder if something is subtly amiss here.

Comment (jian-lin) [reply pkgs/applications/editors/emacs/elisp-packages/melpa-packages.nix:1473]: > My theory is that the underlying error--which this ignores--was recently fixed.

I do not think so.  Many errors reported in https://github.com/drym-org/symex.el/issues/130 are in the build [log].  Those errors are not specific to nix and should be fixed upstream. 

[log]: https://hydra.nixos.org/build/273770270/nixlog/1

**Pull request merged**